packages/syft/src/syft/core/tensor/lazy_repeat_array.py

```diff
@@ -50,87 +60,255 @@ def __init__(self, data: np.ndarray, shape: Tuple[int, ...]) -> None:
         """
         THIS MIGHT LOOK LIKE COPY-PASTED CODE!
         Don't touch it. It's going to get more complicated.
         """
-        if is_acceptable_simple_type(other):
-            return self.__class__(data=self.data - other, shape=self.shape)
-
-        if self.shape != other.shape:
-            raise Exception("cannot subtract tensors with different shapes")
+        if isinstance(other, (int, np.integer, float, np.floating)):
+            return lazyrepeatarray(data=(self.data - other), shape=self.shape)
 
-        if self.data.shape == other.data.shape:
-            return self.__class__(data=self.data - other.data, shape=self.shape)
+        elif isinstance(other, (np.ndarray, lazyrepeatarray)):  # type: ignore
+            if not is_broadcastable(self.shape, other.shape):
+                raise Exception(
+                    f"Cannot broadcast arrays with shapes: {self.shape} & {other.shape}"
+                )
+            self.shape = np.broadcast_shapes(self.shape, other.shape)
+            if isinstance(other, lazyrepeatarray):
+                self.add_op(function="np.subtract", args=other.data)
+            elif isinstance(other, np.ndarray):
+                self.add_op(function="np.subtract", args=other)
+            return self
 
-        raise Exception("not sure how to do this yet")
+        else:
+            raise Exception(f"not sure how to do this yet: {type(other)}")
 
     def __mul__(self, other: Any) -> lazyrepeatarray:
         """
         THIS MIGHT LOOK LIKE COPY-PASTED CODE!
         Don't touch it. It's going to get more complicated.
         """
-        if is_acceptable_simple_type(other):
-            return self.__class__(data=self.data * other, shape=self.shape)
+        if isinstance(other, (int, np.integer, float, np.floating)):
+            return lazyrepeatarray(data=(self.data * other), shape=self.shape)
 
-        if self.shape != other.shape:
-            raise Exception("cannot multiply tensors with different shapes")
+        elif isinstance(other, (np.ndarray, lazyrepeatarray)):  # type: ignore
+            if not is_broadcastable(self.shape, other.shape):
+                raise Exception(
+                    f"Cannot broadcast arrays with shapes: {self.shape} & {other.shape}"
+                )
+            self.shape = np.broadcast_shapes(self.shape, other.shape)
+            if isinstance(other, lazyrepeatarray):
+                self.add_op(function="np.multiply", args=other.data)
+            elif isinstance(other, np.ndarray):
+                self.add_op(function="np.multiply", args=other)
+            return self
 
-        if self.data.shape == other.data.shape:
-            return self.__class__(data=self.data * other.data, shape=self.shape)
+        else:
+            raise Exception(f"not sure how to do this yet: {type(other)}")
 
-        raise Exception("not sure how to do this yet")
+    def __matmul__(self, other: Any):  # type: ignore
```

Comment:
Could we fix the type annotations for `other`, since we know it'll only be a numpy/jnp array, lazyrepeatarray, or integer/float/etc

Reply:
Should we use Union everywhere or declare it inside a class and then import it?